### ISSUE TITLE
CASTRO/wdmerger parameters

### ISSUE THREAD

**Issue by HamidGadirov** [AMReX-Astro/Castro#2974]
Hi,

I am working with CASTRO/wdmerger (`https://github.com/AMReX-Astro/Castro/tree/main/Exec/science/wdmerger`) and wanted to ask which parameters can be changed within the simulation so that to generate an ensemble (runs with different parameters so that to have variations in data).

I am especially interested in density and velocity fields. Is it possible to change parameters so that the data generated by the simulation is a bit different?

Thanks!
Hamid

**Comment by zingale** [comment]
is this and #2966  and #2971 all related?

Can you give some background of what you are trying to accomplish?

Any of the science setups have parameters that can be used to change the initialization.  You should look at the initialization routines to see what they do, and then experiment with some of the parameters that begin with `problem.`

But running a lot of these in 3D can be expensive.

**Comment by HamidGadirov** [comment]
@zingale yes, these issues are related. We are now working with CASTRO/wdmerger and found that `problem.mass_P` and `problem.mass_S` have influence on the dynamics of the data. We tried many other parameters but so far we observe variations in the density and flow fields with these parameters only. Can you let us know if there is another parameter that you recommend to change?

**Comment by khanakbhargava** [comment]
Hi Hamid, 

I would be able to guide you better if you could explain what is it you're trying to do exactly. What are you hoping to achieve from the different set ups? Or are you just trying to learn how the merger works in Castro? 

The easiest thing to do is to adjust the masses of two stars which will calculate the initial densities for you on its own. You can also change the composition of the stars. All parameters you need can be found in [the parameters file](https://github.com/AMReX-Astro/Castro/blob/main/Exec/science/wdmerger/_prob_params). 

I hope this helps.

**Comment by HamidGadirov** [comment]
@khanakbhargava thanks for further assistance on this issue. I am working with colleagues (at the University of Groningen and UC Davis) on the ML-based solution for scientific ensemble visualization. We study how our deep learning-based models can learn new insights from the 3D datasets. For that, we need interesting datasets, that are ensembles (runs with different parameters that result in variations within the dynamics of the data). More specifically, we are interested in the density and velocity fields to study if our models can learn one field form another.

Regarding the` CASTRO/wdmerger`, we see the different dynamics if we change `problem.mass_P `and `problem.mass_S` parameters. Are there any other parameters that would yield different dynamics of the density and velocity fields? Previously, we created an ensemble using Nyx dataset (`https://github.com/AMReX-Astro/Nyx`), details on that are in the paper (`https://arxiv.org/abs/2409.19178`), my colleagues used this dataset for other experiments and papers as well. There, we varied three parameters for the ensemble generation (details are in the paper). We are hoping to achieve a similar ensemble with the `CASTRO/wdmerger` simulation, our intended venues for the publications are TVCG, CGF, and Eurographics.

Best,
Hamid

**Comment by zingale** [comment]
closing this since it seems to be addressed now
